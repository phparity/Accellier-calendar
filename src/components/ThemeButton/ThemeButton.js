import React from "react";

import "./ThemeButton.scss";

/**
 * Renders a themed button with the specified class and text, and handles the onClick event.
 *
 * @param {string} buttonClass - The class for styling the button.
 * @param {string} buttonText - The text to display on the button.
 * @param {function} onClick - The function to be called when the button is clicked.
 * @return {JSX.Element} The themed button component.
 */
function ThemeButton({ buttonClass, buttonText, onClick }) {
  // Create a themed button with the specified class and text
  return (
    <button onClick={onClick} className={`theme_btn ${buttonClass}`}>
      {buttonText}
    </button>
  );
}

export default ThemeButton;
