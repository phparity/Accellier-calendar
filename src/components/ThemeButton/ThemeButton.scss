@import '../../assets/common-style/variable.scss';


.theme_btn{
    font-size: 12px;
    line-height: 14.58px;
    font-family: $font-family-Bold;
    padding: 10px 26px;
    color: $White;
    background-color: $Primary;
    border: none;
    border-radius: 6px;
    -webkit-border-radius: 6px;
    -moz-border-radius: 6px;
    -ms-border-radius: 6px;
    -o-border-radius: 6px;
    transition: .3s;
    -webkit-transition: .3s;
    -moz-transition: .3s;
    -ms-transition: .3s;
    -o-transition: .3s;
    cursor: pointer;
    @media (max-width:1700px){
        font-size: 11px;
        line-height: 13px;
        padding: 5.71px 7.14px 6.29px 7.14px;
        border-radius: 3.57px;
        -webkit-border-radius: 3.57px;
        -moz-border-radius: 3.57px;
        -ms-border-radius: 3.57px;
        -o-border-radius: 3.57px;
    }
    &:hover{
        background-color: $Black;
    }
}

.border_btn{
    font-size: 12px;
    line-height: 14.58px;
    font-family: $font-family-Bold;
    padding: 9px 26px;
    color: $Primary;
    background-color: $White;
    border: 1px solid $Primary;
    border-radius: 6px;
    -webkit-border-radius: 6px;
    -moz-border-radius: 6px;
    -ms-border-radius: 6px;
    -o-border-radius: 6px;
    transition: .3s;
    -webkit-transition: .3s;
    -moz-transition: .3s;
    -ms-transition: .3s;
    -o-transition: .3s;
    cursor: pointer;
    @media (max-width:1700px) {
        font-size: 11px;
        line-height: 13px;
        padding: 4.71px 7.14px 5.29px 7.14px;
        border-radius: 3.57px;
        -webkit-border-radius: 3.57px;
        -moz-border-radius: 3.57px;
        -ms-border-radius: 3.57px;
        -o-border-radius: 3.57px;
    }
    &:hover{
        color: $White;
        background-color: $Primary;
    }
}