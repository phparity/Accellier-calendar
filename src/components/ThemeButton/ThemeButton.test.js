import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import ThemeButton from "./ThemeButton";

test("ThemeButton renders properly and calls onClick function when clicked", () => {
  const mockOnClick = jest.fn();

  const { getByText } = render(
    <ThemeButton
      buttonClass="test-class"
      buttonText="Click Me"
      onClick={mockOnClick}
    />,
  );

  const button = getByText("Click Me");
  expect(button).toBeInTheDocument();

  fireEvent.click(button);

  expect(mockOnClick).toHaveBeenCalled();
});

describe("Theme button Component", () => {
  it("renders correctly", () => {
    const mockOnClick = jest.fn();
    const tree = renderer
      .create(
        <ThemeButton
          buttonClass="test-class"
          buttonText="Click Me"
          onClick={mockOnClick}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
