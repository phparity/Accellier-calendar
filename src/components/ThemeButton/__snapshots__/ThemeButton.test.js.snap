// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme button Component renders correctly 1`] = `
<button
  className="theme_btn test-class"
  onClick={[MockFunction]}
>
  Click Me
</button>
`;
